### PR TITLE
fix:update product using formdata

### PR DESCRIPTION
**Pull request by JENNY-257**
What does this PR do?

A seller should be able to update product from thier selle's collection

How should this be manually tested?

Installation
1. Clone the repository by running

git clone https://github.com/atlp-rwanda/destructors-ec-bn.git

2. Navigate to the project directory by running cd destructors-ec-bn.

cd destructors-ec-bn
 cd  bg-fix-product-availability-184637144

3. Install the necessary dependencies by running npm install.

npm install
4. run migrations

 npx sequelize-cli db:migrate
5. run seeds

npx sequelize-cli db:seed:all

6. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.

npm run dev

7. Run the tests by running npm run test

npm run test

Any background context you want to provide?

when running this on a local server put into consideration the below information 

 once you are a seller add the seller sign in  token in headers with a key of Authorization and make sure that the value is started by Bearer

for  updating product end point PATCH: /api/v1/products/{id}
Questions:
none

### CODE REVIEW

File: src/controllers/product.controller.js [hunk 1]
```diff
@@ -273,15 +273,12 @@ const updateProduct = async (req, res) => {
     }
     const { name, price, quantity, bonus, expiryDate, categoryId } = req.body;
     await Products.update(
-      { name, price, quantity, categoryId, bonus, expiryDate },
+      { name, price, quantity, categoryId, bonus, expiryDate,images },
       { where: { id, sellerId: seller.id } }
```
Comment (katros1) [inline src/controllers/product.controller.js]: was it necessary to force seller to update  all fields of the Products like name, price, quantity, bonus etc all at the once whenever he/she want to make an update??? I think seller should update the some fields of his/her choice without changing all of them!!!

Comment (JENNY-257) [reply src/controllers/product.controller.js]: thanks @katros1 , fixed

File: src/routes/api/products.routes.js [hunk 2]
```diff
@@ -42,11 +42,9 @@ route.patch(
   checkRole(['seller']),
   updateProductAvailability
 );
-route.patch('/:id', extractToken, checkRole(['seller']),checkIfPasswordIsExpired, updateProduct);
-route.delete('/:id', extractToken, checkRole(['seller']),checkIfPasswordIsExpired, deleteProduct);
-route.get('/:id/product-wishes',extractToken,getWishesPerProduct)
-route.patch('/:id', extractToken, checkRole(['seller']), updateProduct);
-route.delete('/:id', extractToken, checkRole(['seller']), deleteProduct);
-route.post('/:id/reviews',extractToken,checkRole(['buyer']),validateReviewProduct,createProductReview)
+route.patch('/:id', extractToken, checkRole(['seller']), uploadArray('image'), updateProduct, checkIfPasswordIsExpired);
```
Comment (katros1) [inline src/routes/api/products.routes.js]: A middleware should come before controller; change this checkIfPasswordIsExpired middleware position

**Review by katros1 — CHANGES_REQUESTED** [review]

**Review by ba-lambert — APPROVED** [review]
But it would be better if you squash your commits

**Review by katros1 — APPROVED** [review]

**Review by Abdulkeza — APPROVED** [review]

**Pull request merged**